core/utils.js

```diff
@@ -99,33 +99,39 @@ Blockly.hasClass_ = function(element, className) {
  * @private
  */
 Blockly.bindEvent_ = function(node, name, thisObject, func) {
-  if (thisObject) {
-    var wrapFunc = function(e) {
-      func.call(thisObject, e);
-    };
-  } else {
-    var wrapFunc = func;
-  }
+  var wrapFunc = function(e) {
+    // Handle each touch point separately.  If the event was a mouse event, this
+    // will hand back an array with one element, which we're fine handling.
+    var events = Blockly.bindEvent_.splitEventByTouches(e);
+    for (var i = 0; i < events.length; i++) {
+      var event = events[i];
+      if (!Blockly.checkTouchIdentifier(event)) {
+        return;
+      }
+      Blockly.bindEvent_.setClientFromTouch(event);
```

Comment:
I wonder if it might actually make sense to only do this when a logical "drag" has started (e.g., once DRAG_RADIUS has been exceeded for a block drag). That way you could put a bunch of fingers on blocks, but only disabled touches for most of them once the first block starts dragging. Might be best to save that for a later PR though? Thoughts on the idea?